src/components/Firestore/DocumentEditor/index.tsx

```diff
@@ -302,15 +325,20 @@ const FieldEditor: React.FC<{
 const ChildTypeSelect: React.FC<{
   uuid: number;
   isRtdb: boolean;
-}> = ({ uuid, isRtdb }) => {
+  blacklist?: FieldType[];
+}> = ({ uuid, isRtdb, blacklist = [] }) => {
   const field = useField(uuid);
   const dispatch = useDispatch();
 
+  const options = (isRtdb ? RTDB_FIELD_TYPES : FIRESTORE_FIELD_TYPES).filter(
```

Comment:
Can this be memo'd for less re-renders? it'll only enumerate to 3 or 4 possibilities ever

Reply:
Pulled these out, and now just storing the "ARRAY_FIELD_TYPES" in a const outside of the component.